xapian-core/weight/coordweight.cc

```diff
@@ -86,4 +92,12 @@ CoordWeight::get_maxextra() const
     return 0;
 }
 
+CoordWeight *
+CoordWeight::create_from_parameters(const char * p) const
+{
+    if (*p == ' ')
+	throw InvalidArgumentError("No parameters are required for CoordWeight");
+    return new Xapian::CoordWeight();
```

Comment:
This is the same situation as `BoolWeight` (i.e. there shouldn't be any parameters), but this is handled differently - differently wrongly.  The question that's relevant is whether the string is empty or not, not whether it starts with a space.